controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/Versions.java

```diff
@@ -70,6 +70,28 @@ public boolean targetsMatch(JobStatus.JobRun jobRun) {
                targetApplication.equals(jobRun.application());
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if ( ! (o instanceof Versions)) return false;
+
+        Versions versions = (Versions) o;
+
+        if ( ! targetPlatform.equals(versions.targetPlatform)) return false;
+        if ( ! targetApplication.equals(versions.targetApplication)) return false;
+        if ( ! sourcePlatform.equals(versions.sourcePlatform)) return false;
+        return sourceApplication.equals(versions.sourceApplication);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = targetPlatform.hashCode();
```

Comment:
PS: Objects.hash(...)

Reply:
Yes, that's cleaner. I'll change the IntelliJ settings ...